src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -1,146 +1,188 @@
 import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
 import ch.njol.skript.registrations.Classes;
+import ch.njol.util.Kleenean;
 import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
 import org.bukkit.event.inventory.InventoryType;
 import org.bukkit.inventory.Inventory;
 import org.jetbrains.annotations.Nullable;
 
-import ch.njol.skript.Skript;
-import ch.njol.skript.doc.Description;
-import ch.njol.skript.doc.Examples;
-import ch.njol.skript.doc.Name;
-import ch.njol.skript.doc.Since;
-import ch.njol.skript.lang.Effect;
-import ch.njol.skript.lang.Expression;
-import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.util.Kleenean;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
 
 @Name("Open/Close Inventory")
-@Description({"Opens an inventory to a player. The player can then access and modify the inventory as if it was a chest that he just opened.",
-		"Please note that currently 'show' and 'open' have the same effect, but 'show' will eventually show an unmodifiable view of the inventory in the future."})
-@Examples({"show the victim's inventory to the player",
-		"open the player's inventory for the player"})
-@Since("2.0, 2.1.1 (closing), 2.2-Fixes-V10 (anvil), 2.4 (hopper, dropper, dispenser")
+@Description("""
+	Opens an inventory to a player. The player can then access and modify the inventory as if it was a chest that they opened.
+	Please note that currently 'show' and 'open' have the same effect,\
+	but 'show' will eventually show an unmodifiable view of the inventory in the future.
+	""")
+@Example("open the player's inventory for the player")
+@Example("open an anvil window for all players")
+@Example("show grindstone view to {_player}")
+@Example("open a dropper inventory for player")
+@Example("close the inventory for all players")
+@Since("2.0, 2.1.1 (closing), 2.2-Fixes-V10 (anvil), 2.4 (hopper, dropper, dispenser)")
 public class EffOpenInventory extends Effect {
-	
-	private final static int WORKBENCH = 0, CHEST = 1, ANVIL = 2, HOPPER = 3, DROPPER = 4, DISPENSER = 5;
+
+	// TODO: Add support for Paper's MenuType (1.21.3+)
+
+	private enum OpenableType {
+		ANVIL("[an] anvil", "anvil", InventoryType.ANVIL),
+		CARTOGRAPHY("[a] cartography [table]", "cartography table", InventoryType.CARTOGRAPHY),
+		CRAFTING("[a] (crafting [table]|workbench)", "crafting table", InventoryType.CRAFTING),
+		DISPENSER("dispenser", InventoryType.DISPENSER),
+		DROPPER("dropper", InventoryType.DROPPER),
+		ENCHANTING("[an] enchant(ing|ment) [table]", "enchantment table", InventoryType.ENCHANTING),
+		GRINDSTONE("grindstone", InventoryType.GRINDSTONE),
+		HOPPER("hopper", InventoryType.HOPPER),
+		LOOM("loom", InventoryType.LOOM),
+		SMITHING("[a] smithing [table]", "smithing table", InventoryType.SMITHING),
+		STONECUTTER("stonecutter", InventoryType.STONECUTTER)
```

Comment:
Isn't this what the `inventorytype` type handles already? Can't we just improve it instead of duplicating it like this?